claim_batch/services.py

```diff
@@ -546,91 +569,108 @@ def do_process_batch(audit_user_id, location_id, period, year):
 def get_allocated_premium(premiums, start_date, end_date):
     # go trough the contribution and find the allocated contribution 
     allocated_premiums = 0
-    for premiums in premium:
-        allocation_start = max(policy.effective_date, start_date)
-        allocation_stop = min(start_date, policy.expiry_date)
-        policy_duration = policy.expiry_date(  - policy.effective_date 
-        allocated_premiums += premium.amount * (allocation_stop - allocation_start ) / policy_duration
-    return  allocated_premiums
+    for premium in premiums:
+        allocation_start = max(premium.policy.effective_date, start_date)
+        allocation_stop = min(start_date, premium.policy.expiry_date)
+        policy_duration = premium.policy.expiry_date - premium.policy.effective_date
+        allocated_premiums += premium.amount * (allocation_stop - allocation_start) / policy_duration
+    return allocated_premiums
+
+
+def get_period(start_date, end_date):
+    # TODO do function that returns such values M/Q/Y , 1-12/1-4/1
+    pass
 
    
 # update the service and item valuated amount 
 def claim_batch_valuation(work_data, start_date, end_date):
-    allocated_contributions = get_allocated_premium(work_data.contributions, start_date, end_date)
+    allocated_contributions = work_data["allocated_contributions"]
     # Sum up all item and service amount
     value_hospital = 0
     value_non_hospital = 0
-    periode_type, period_id = get_period(start_date, end_date); # M/Q/Y , 1-12/1-4/1
+    period_type, period_id = get_period(start_date, end_date)
+
+    product = work_data["product"]
+    items = work_data["items"]
+    services = work_data["services"]
+
     # ensure that the product need to be valuated for the given period
-    if period_rel_prices  == periode_type:
-        period_rate = get_relative_price_rate(work_data.product, 'B' periode_type, period_id) # migh be added in product service
-    elif period_rel_prices_ip == periode_type:
-        period_rate_ip = get_relative_price_rate(work_data.product, 'I' periode_type, period_id) # migh be added in product service
-    elif period_rel_prices_op == periode_type
-        period_rate_op = get_relative_price_rate(work_data.product, 'O' periode_type, period_id) # migh be added in product service
-
-    if period_rate>0 or period_rate_ip>0 or period_rate_op>0:
+    if product.period_rel_prices == period_type:
+        period_rate = get_relative_price_rate(product, 'B', period_type, period_id) # migh be added in product service
+    elif product.period_rel_prices_ip == period_type:
```

Comment:
this should be an else, with a 2 ifs inside it (not if /else)

Reply:
ok noted. @delcroip . I will fix this